CMakeLists.txt

```diff
@@ -56,6 +56,17 @@ set(CMAKE_CXX_CREATE_SHARED_LIBRARY)
 set(CMAKE_DEBUG_POSTFIX "d")
 set(CMAKE_RELEASE_POSTFIX "r")
 
+# Set a default build type if none was specified
+set(default_build_type "Release")
+if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+  message(STATUS "Setting build type to '${default_build_type}' as none was specified.")
+  set(CMAKE_BUILD_TYPE "${default_build_type}" CACHE
+      STRING "Choose the type of build." FORCE)
+  # Set the possible values of build type for cmake-gui
+  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS
```

Comment:
```suggestion
  set_property(CACHE CMAKE_BUILD_TYPES PROPERTY STRINGS
```

Reply:
Could be wrong, but it looks strange at least.

Reply:
Seems fine from the docs: https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html